### ISSUE TITLE
Unexpected semantics difference between flutter test and flutter run

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#19181]
The following test case passes when executed with flutter test:

```dart
  testWidgets('has only root node if surface size is 0x0', (WidgetTester tester) async {
    final SemanticsTester semantics = new SemanticsTester(tester);

    await tester.pumpWidget(new Semantics(
      selected: true,
    ));

    debugDumpRenderTree();

    expect(semantics, hasSemantics(
      new TestSemantics(
        id: 0,
        rect: new Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0),
        children: <TestSemantics>[
          new TestSemantics(
            id: 1,
            rect: new Rect.fromLTRB(0.0, 0.0, 800.0, 600.0),
            flags: <SemanticsFlag>[SemanticsFlag.isSelected],
          ),
        ],
      ), ignoreTransform: true,
    ));

    semantics.dispose();
  });
```

But it fails when executed with `flutter run`:
```
flutter:   Expected: semantics node matching:
flutter:           new TestSemantics(
flutter:             id: 0,
flutter:             rect: Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0),
flutter:             children: <TestSemantics>[
flutter:               new TestSemantics(
flutter:                 id: 1,
flutter:                 flags: <SemanticsFlag>[SemanticsFlag.isSelected],
flutter:                 rect: Rect.fromLTRB(0.0, 0.0, 800.0, 600.0),
flutter:                 children: <TestSemantics>[
flutter:                 ],
flutter:               ),
flutter:             ],
flutter:           )
flutter:   Actual: SemanticsTester:<SemanticsTester for SemanticsNode#0(Rect.fromLTRB(0.0, 385.8, 750.0,
flutter:   948.3))>
flutter:    Which: expected node id 0 to have rect Rect.fromLTRB(0.0, 0.0, 2400.0, 1800.0) but found rect
flutter:    Rect.fromLTRB(0.0, 385.8, 750.0, 948.3).
flutter:           Current SemanticsNode tree:
flutter:           SemanticsNode#0
flutter:            │ Rect.fromLTRB(0.0, 385.8, 750.0, 948.3)
flutter:            │
flutter:            └─SemanticsNode#1
flutter:                Rect.fromLTRB(0.0, 0.0, 800.0, 600.0) with transform
flutter:                  [0.9375,0.0,0.0,0.0; 0.0,0.9375,0.0,385.75; 0.0,0.0,1.0,0.0;
flutter:                  0.0,0.0,0.0,1.0]
flutter:                flags: isSelected
flutter:           The semantics tree would have matched the following configuration:
flutter:           new TestSemantics.root(
flutter:             children: <TestSemantics>[
flutter:               new TestSemantics(
flutter:                 id: 1,
flutter:                 flags: <SemanticsFlag>[SemanticsFlag.isSelected],
flutter:               ),
flutter:             ],
flutter:           )
```

Looks like the root semantics node has an unexpected size when doing a `flutter run`.

**Comment by goderbauer** [comment]
The difference in window size (and pixel ratio) is also present in the render tree:

From `flutter test`:

```
RenderView#4c7a9
 │ debug mode enabled - macos
 │ window size: Size(2400.0, 1800.0) (in physical pixels)
 │ device pixel ratio: 3.0 (physical pixels per logical pixel)
 │ configuration: Size(800.0, 600.0) at 3.0x (in logical pixels)
 │
 └─child: RenderSemanticsAnnotations#840c1
     creator: Semantics ← [root]
     parentData: <none>
     constraints: BoxConstraints(w=800.0, h=600.0)
     size: Size(800.0, 600.0)
```

From `flutter run`:

```
flutter: _LiveTestRenderView#61753
flutter:  │ debug mode enabled - ios
flutter:  │ window size: Size(750.0, 1334.0) (in physical pixels)
flutter:  │ device pixel ratio: 2.0 (physical pixels per logical pixel)
flutter:  │ configuration: TestViewConfiguration (in logical pixels)
flutter:  │
flutter:  └─child: RenderSemanticsAnnotations#d3bb0
flutter:      creator: Semantics ← [root]
flutter:      parentData: <none>
flutter:      constraints: BoxConstraints(w=800.0, h=600.0)
flutter:      size: Size(800.0, 600.0)
```

**Comment by maksimr** [comment]
@goderbauer is there any way to control RenderView(window) size in tests?

Thanks

**Comment by zoechi** [comment]
@maksimr I only found 
```
final TestViewConfiguration small =
    TestViewConfiguration(size: const Size(600.0, 800.0));

WidgetsBinding.instance.renderView.configuration = small;
```
and it somehow changes the view but I can't claim to have understood how exactly it works.
It doesn't have influence on MediaQuery orientation

**Comment by maksimr** [comment]
@zoechi it works perfectly for me. Thanks!

**Comment by goderbauer** [comment]
This is not part of our public API and even within the framework we don not recommend the use of TestSemantics anymore.
